### PR TITLE
nixos: improve "enable" descriptions

### PR DESCRIPTION
**Pull request by bjornfor**
## Description of changes

- nixos/ulogd: improve 'enable' description
- nixos/mysql: improve 'enable' description
- nixos/xdg: improve 'enable' description
[...snip ~100 commits...]

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/programs/wayland/hyprland.nix [hunk 1]
```diff
@@ -19,7 +19,7 @@ in
         You can manually launch Hyprland by executing {command}`Hyprland` on a TTY.
 
         A configuration file will be generated in {file}`~/.config/hypr/hyprland.conf`.
-        See <https://wiki.hyprland.org> for more information.
+        See <https://wiki.hyprland.org> for more information
```
Comment (emilylange) [inline nixos/modules/programs/wayland/hyprland.nix:22]: Again, merging.

**Review by emilylange — CHANGES_REQUESTED** [review]
I am also not all that convinced this needs 109 separate commits.
This is a tree-wide change.

IMHO this should be squashed.

You could also drop `lib.mdDoc` as part of this change, since it is no longer required, if you like.

And you could further improve this by using markdown/commonmark href links (`[text](https://example.com)`) to link to each project.

FYI: You can build the manual locally using `nix-build nixos/release.nix -A manual.x86_64-linux` and then open `result/share/doc/nixos/options.html` in your browser of choice.

File: nixos/modules/programs/regreet.nix [hunk 2]
```diff
@@ -11,7 +11,7 @@ in
   options.programs.regreet = {
     enable = lib.mkEnableOption null // {
       description = lib.mdDoc ''
-        Enable ReGreet, a clean and customizable greeter for greetd.
+        ReGreet, a clean and customizable greeter for greetd.
```
[the remark this answers is not in the document]
Comment (bjornfor) [reply nixos/modules/programs/regreet.nix]: Fixed, thanks!

File: nixos/modules/programs/minipro.nix [hunk 3]
```diff
@@ -8,8 +8,8 @@ in
     programs.minipro = {
       enable = lib.mkEnableOption (lib.mdDoc "minipro") // {
         description = lib.mdDoc ''
-          Installs minipro and its udev rules.
-          Users of the `plugdev` group can interact with connected MiniPRO chip programmers.
+          minipro and its udev rules.
+          Users of the `plugdev` group can interact with connected MiniPRO chip programmers
```
Comment (emilylange) [inline nixos/modules/programs/minipro.nix:12]: Has the merging thingy again.
Looks like I have missed that in my initial review.

```suggestion
          Whether to enable minipro and its udev rules.
          Users of the `plugdev` group can interact with connected MiniPRO chip programmers.
```

**Review by emilylange — CHANGES_REQUESTED** [review]
Just one more thing, LGTM otherwise :)

File: nixos/modules/programs/minipro.nix (hunk 3 above)
Comment (bjornfor) [reply nixos/modules/programs/minipro.nix]: Fixed, thanks!

**Review by emilylange — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Comment by bjornfor** [comment]
Rebased to resolve conflict. I plan to merge after the CI run.

**Pull request merged**